src/global/project/mod.rs

```diff
@@ -80,31 +105,257 @@ impl Project {
         Ok(Self::from_manifest(manifest))
     }
 
-    /// Discovers the project manifest file in path set by `PIXI_GLOBAL_MANIFESTS`
-    /// or alternatively at `~/.pixi/manifests/pixi-global.toml`.
-    /// If the manifest doesn't exist yet, and empty one will be created.
-    pub(crate) fn discover() -> miette::Result<Self> {
+    /// Discovers the project manifest file in path at `~/.pixi/manifests/pixi-global.toml`.
+    /// If the manifest doesn't exist yet, and the function will try to create one from the existing installation.
+    /// If that one fails, an empty one will be created.
+    pub(crate) async fn discover(
```

Comment:
small nitpick - I think we can name it `discover_or_create`

Reply:
Your suggested name sounds good to me, changed!